azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -0,0 +1,494 @@
+            keyvault_name=spc_keyvault_name,
+        )
+
+        secrets: PageIterator = self.keyvault_client.list_properties_of_secrets()
+
+        # process all the file validations before secret creations
+        self._validate_key_files(public_key_file, private_key_file)
+
+        # check if there is a spc called "opc-ua-connector", if not create one
+        try:
+            opcua_spc = self.ssc_mgmt_client.azure_key_vault_secret_provider_classes.get(
+                resource_group_name=resource_group,
+                azure_key_vault_secret_provider_class_name=OPCUA_SPC_NAME,
+            )
+        except ResourceNotFoundError:
+            opcua_spc = {}
+
+        # check if there is a secret sync called "aio-opc-ua-broker-client-certificate", if not create one
+        try:
+            opcua_secret_sync = self.ssc_mgmt_client.secret_syncs.get(
+                resource_group_name=resource_group,
+                secret_sync_name=OPCUA_CLIENT_CERT_SECRET_SYNC_NAME,
+            )
+        except ResourceNotFoundError:
+            opcua_secret_sync = {}
+
+        secrets_to_add = []
+        for file in [public_key_file, private_key_file]:
+            file_name = os.path.basename(file)
+            file_name_info = os.path.splitext(file_name)
+            cert_extension = file_name_info[1].replace(".", "")
+            cert_name = file_name_info[0].replace(".", "-")
+            secret_name = f"{cert_name}-{cert_extension}"
+
+            # iterate over secrets to check if secret with same name exists
+            secret_name = self._check_and_update_secret_name(secrets, secret_name, spc_keyvault_name)
+            self._upload_to_key_vault(secret_name, file, cert_extension)
+            secrets_to_add.append((secret_name, file_name))
+
+        self._add_secrets_to_spc(
+            secrets=[secret[0] for secret in secrets_to_add],
+            spc=opcua_spc,
+            resource_group=resource_group,
+            spc_keyvault_name=spc_keyvault_name,
+            spc_tenant_id=spc_tenant_id,
+            spc_client_id=spc_client_id,
+        )
+
+        self._add_secrets_to_secret_sync(
+            secrets=secrets_to_add,
+            secret_sync=opcua_secret_sync,
+            resource_group=resource_group,
+            spc_name=OPCUA_SPC_NAME,
+            secret_sync_name=OPCUA_CLIENT_CERT_SECRET_SYNC_NAME,
+        )
+
+        # update opcua extension
+        return self._update_client_secret_to_extension(
+            subject_name=subject_name,
+            application_uri=application_uri,
+        )
+
+    def _validate_key_files(self, public_key_file: str, private_key_file: str):
+        # validate public key file end with .der
+        validate_file_extension(public_key_file, [".der"])
+        # validate private key file end with .pem
+        validate_file_extension(private_key_file, [".pem"])
+
+        # validate public key and private key has matching file name without extension
```

Comment:
is this a service constraint?

Reply:
yeah service team would want us to check this beforehead, because it involve multi steps so they want the issue to be validate earlier before it's not working at the very end